en/docs/develop/manage-repository/connect-your-own-github-repository-to-choreo.md

```diff
@@ -1,148 +1,69 @@
       |-----------------|------------------------------|
       | **Name**        | COVID-19 Statistics          |
       | **Description** | Maintain COVID-19 Statistics |
 
-4. Click **Create**. This takes you to the **Components** page.
+4. Click **Create**. 
+
+## Connect your GitHub repository
+
+Let's connect your GitHub repository and use the REST API implementation in it to create a REST API component in Choreo: 
 
-## Step 2: Add a REST API component and connect it to your own GitHub repository
+1. On the **Components** page, click **+Create** on the the **REST API** card.
 
-1. On the **Components** page, click **+Create**.
-   ![Create component](../../assets/img/tutorials/connect-own-repo/create-component.png){.cInlineImage-full}
-2. Click the **REST API** card.
-3. Click **Create an API from scratch** and enter a unique name and a description for the API. In this tutorial, let's enter the following values:
+    ![Create component](../../assets/img/tutorials/connect-own-repo/create-component.png){.cInlineImage-small}
+
+2. Enter a unique name and a description for the API. In this tutorial, let's enter the following values:
 
       | **Field**       | **Value**                   |
       |-----------------|-----------------------------|
       | **Name**        | COVID-19 Statistics         |
       | **Description** | Retrieve COVID-19 Statistics|
       
-4. Click **Next**.
-5. Select **Connect your own GitHub repository**.
-6. To authorize Choreo to access your GitHub account, click **Continue**, enter your GitHub credentials, and select one or more repositories to install the [Choreo GitHub App](https://github.com/marketplace/choreo-apps).
-
-    ![Authorize GitHub app](../../assets/img/tutorials/connect-own-repo/authorize-github-app.png){.cInlineImage-half}
-
-    !!!info
-
-         - The **Choreo GitHub App** requires the following permission:
-            - Read and write access to code and pull requests.
-            - Read access to issues and metadata.
-         - You can [revoke access](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/reviewing-your-authorized-integrations#reviewing-your-authorized-github-apps) if you do not want Choreo to have access to your GitHub account. However, write access is only used to send pull requests to a user repository. Choreo will not directly push any changes to a repository.
-
-7. Select a GitHub account and a GitHub repository that includes a **Choreo GitHub App** installation.
-
-    !!! tip 
-        Choreo does not allow you to connect GitHub repositories that include existing Ballerina projects.
+3. Click **Next**.
+4. Provide authorization with your GitHub account by clicking  **Authorize with GitHub**. 
+5. If you have not already authorized Choreo apps, click **Authorize Choreo Apps** when prompted.
+6. Select a GitHub account, a repository that includes the implementation, the relevant branch, and the build preset: Ballerina or Dockerfile. 
```

Comment:
The build preset has to be Ballerina in this case, isn't it?

Reply:
Done.
@indikasampath2000  we are not taking them through a particular sample. Therefore they can actually bring in any implementation to complete this tutorial.

Reply:
Maybe it is better to instruct the users to select Ballerina for this example because the fields that appear after this depends on what the user selected. If the user selects **Docker** at this point, he/she will expect to see steps to enter the Dockerfile path, context, etc.

Reply:
Until we get the implementation-related details sorted, shall we leave it as it is? Because I think without having a sample, it doesn't add any value by asking a user to select a specific preset. It could lead to more questions.

Reply:
Then again, someone who happens to select **Docker** would think that the steps after this are wrong and report issues.

Then we might need to fork the next step to match either scenario...like "If you selected **Ballerina** as the build preset, enter a valid path relative to the root of your repository that points to the implementation of the REST API. If you selected **Docker**, etc., etc.,...."

Reply:
I guess that's a secondary issue. Let's deal with it when we get there. Anyway we would decide on the way forward with this in the coming few weeks. So, I guess until then it's fine.